packages/auth/src/handlers.ts

```diff
@@ -128,24 +132,52 @@ export const requests = {
       throw new Error(`No user found with email: ${email}`)
     }
 
-    const groupUpdateData = {
-      schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
-      Operations: [
-        {
-          op: operation,
-          path: 'members',
-          value: [
+    if (operation === Wso2GroupOperation.Add) {
+      const hasSponsorEngagementTool =
+        Array.isArray(user.groups) &&
+        user.groups.some((group) => typeof group === 'object' && group.display === ODP_ROLE)
+
+      if (hasSponsorEngagementTool) {
+        return
+      }
+    }
+
+    let groupUpdateData: GroupUpdateData
+
+    switch (operation) {
+      case Wso2GroupOperation.Add:
+        groupUpdateData = {
+          schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
+          Operations: [
             {
-              display: user.userName, // The user's username
-              value: user.id, // The user's SCIM ID
+              op: Wso2GroupOperation.Add,
+              value: {
+                members: [
+                  {
+                    display: user.userName, // The user's username
+                    value: user.id, // The user's SCIM ID
+                  },
+                ],
+              },
             },
           ],
-        },
-      ],
-    }
+        }
+        break
 
-    const response = await api.patch(`/scim2/Groups/${group}`, groupUpdateData)
+      case Wso2GroupOperation.Remove:
+        groupUpdateData = {
+          schemas: ['urn:ietf:params:scim:api:messages:2.0:PatchOp'],
+          Operations: [
+            {
+              op: Wso2GroupOperation.Remove,
+              path: `members[value eq ${user.id}]`, // The user's SCIM ID
+            },
+          ],
+        }
+        break
+    }
 
+    const response = await requests.patchUserGroup(groupId, groupUpdateData)
```

Comment:
I think we talked about checking response codes here and adding some logging to enable debugging in the future

Reply:
added catch

Reply:
I wouldn't expect an exception to be thrown if a non success status code was returned (we saw the 400 - but no unhandled exception right) - so I think we still need to add some logic to get the status code and log it.... correct me if I am wrong here